control_plane/src/endpoint.rs

```diff
@@ -169,6 +170,30 @@ impl ComputeControlPlane {
 
         Ok(ep)
     }
+
+    pub fn check_conflicting_endpoints(
+        &self,
+        mode: ComputeMode,
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+    ) -> Result<()> {
+        if matches!(mode, ComputeMode::Primary) {
+            // this check is not complete, as you could have a concurrent attempt at
+            // creating another primary, both reading the state before checking it here,
+            // but it's better than nothing.
+            let mut duplicates = self.endpoints.iter().filter(|(_k, v)| {
+                v.tenant_id == tenant_id
+                    && v.timeline_id == timeline_id
+                    && v.mode == mode
+                    && v.status().starts_with("running")
```

Comment:
should we also check not only for "running" endpoints, but also for "crashed" (we cannot connect, but pid still exists), because this may mean that some background processes are still running ?

Reply:
So basically `v.status() == "stopped"`?

Reply:
> should we also check not only for "running" endpoints, but also for "crashed" (we cannot connect, but pid still exists)

Such an endpoint would not be of any use to a client(because can't connect) and will block the creation of another endpoint(as a replacement of "crashed" endpoint)

Reply:
I've been running with the `v.status() != "stopped"` the `test_branching_with_pgbench` without any luck so far.

Reply:
```suggestion
                    && v.status() != "stopped"
```